tests/Unit/Evolution/Systems/CurvedScalarWave/Worldtube/TestHelper.hpp

```diff
@@ -0,0 +1,110 @@
+          tmpl::pair<DomainCreator<3>,
+                     tmpl::list<::domain::creators::BinaryCompactObject>>>;
+    };
+  };
+
+  // returns a binary compact object domain creator with the excision spheres
+  // corresponding to the circular worldtube setup: a central excision sphere
+  // (the central black hole) and a worldtube excision sphere in circular orbit
+  // around it with angular velocity R^{-3/2}, where R is the orbital radius.
+  static std::unique_ptr<DomainCreator<3>> worldtube_binary_compact_object(
+      const double orbit_radius, const double worldtube_radius) {
+    ASSERT(orbit_radius > 4. * worldtube_radius,
+           "Can't construct a domain with these parameters");
+    const double angular_velocity = 1. / (orbit_radius * sqrt(orbit_radius));
+    std::string binary_compact_object_options =
+        "BinaryCompactObject:\n"
+        "  ObjectA:\n"
+        "    InnerRadius: " +
+        std::to_string(worldtube_radius) +
+        "\n"
+        "    OuterRadius: " +
+        std::to_string(2. * worldtube_radius) +
+        "\n"
+        "    XCoord: " +
+        std::to_string(orbit_radius) +
+        "\n"
+        "    Interior:\n"
+        "      ExciseWithBoundaryCondition:\n"
+        "        None:\n"
+        "    UseLogarithmicMap: false\n"
+        "  ObjectB:\n"
+        "    InnerRadius: 1.9\n"
+        "    OuterRadius: 3.0\n"
+        "    XCoord: -1e-99\n"
+        "    Interior:\n"
+        "      ExciseWithBoundaryCondition:\n"
+        "        None\n"
+        "    UseLogarithmicMap: false\n"
+        "  EnvelopingCube:\n"
+        "    Radius: 30.0\n"
+        "    UseProjectiveMap: false\n"
+        "    Sphericity: 1.0\n"
+        "  OuterShell:\n"
+        "    InnerRadius: Auto\n"
+        "    OuterRadius: 50.0\n"
+        "    RadialDistribution: Linear\n"
+        "    BoundaryCondition:\n"
+        "      None\n"
+        "  InitialRefinement:\n"
+        "    ObjectAShell: [0, 0, 0]\n"
+        "    ObjectBShell: [0, 0, 0]\n"
+        "    ObjectACube: [0, 0, 0]\n"
+        "    ObjectBCube: [0, 0, 0]\n"
+        "    EnvelopingCube: [0, 0, 0]\n"
+        "    OuterShell: [0, 0, 0]\n"
+        "  InitialGridPoints: 3\n"
+        "  UseEquiangularMap: true\n"
+        "  TimeDependentMaps:\n"
+        "    InitialTime: 0.0\n"
+        "    ExpansionMap: \n"
+        "      OuterBoundary: 1.0\n"
+        "      InitialExpansion: 1.0\n"
+        "      InitialExpansionVelocity: 0.0\n"
+        "      AsymptoticVelocityOuterBoundary: 0.0\n"
+        "      DecayTimescaleOuterBoundaryVelocity: 1.0\n"
+        "    RotationMap:\n"
+        "      InitialAngularVelocity: [0.0, 0.0," +
+        std::to_string(angular_velocity) +
+        "]\n"
+        "    SizeMap:\n"
+        "      InitialValues: [0.0, 0.0]\n"
+        "      InitialVelocities: [0.0, 0.0]\n"
+        "      InitialAccelerations: [0.0, 0.0]";
+    return TestHelpers::test_option_tag<domain::OptionTags::DomainCreator<3>,
```

Comment:
Just call the constructor instead of going through option parsing?

Reply:
I found this a lot easier to read because you can immediately see exactly what goes into the constructor without looking up all the dozens of arguments at each position if that is fine with you

Reply:
That C++ doesn't support named arguments is a complaint you have to take up with the C++ committee :D Maybe C++20 will improve things a little with designated initializers.

I guess option creation is fine if you prefer it.